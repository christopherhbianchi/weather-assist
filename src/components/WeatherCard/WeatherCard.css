img{
  display: block;
  margin-left: auto;
  margin-right: auto;
  max-width: 30%;
}
.weather-card{
  background: #43484a;
  color: white;
  border-radius: 10px;
  padding-top: 5px;
  padding-bottom: 15px;
  max-width: 50%;
  margin: auto;
  box-shadow: 8px 18px 47px #0e0e0e;
}
.cityName{
  margin-top: 0;
  margin-bottom: 15px;
  font-weight: 300;
  color: #e7e8e8;
}

.remove-button {
  background: none;
  background-image: url('../../assets/blueX.png');
  background-size: 18px;
  background-repeat: no-repeat;
  height: 24px;
  width: 25px;
  border: none;
  display: block;
  margin-top: 3px;
  margin-left: auto;
}

.remove-button:hover {
  background: none;
  background-image: url('../../assets/redX.png');
  cursor: pointer;
  background-size: 18px;
  background-repeat: no-repeat;
  height: 24px;
  width: 25px;
  border: none;
  display: block;
  margin-top: 3px;
  margin-left: auto;
}

p{
  font-size: 12px;
  font-weight: 200;
  color: #e6e7e7;
}
