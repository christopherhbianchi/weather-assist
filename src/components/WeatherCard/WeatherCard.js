import React from 'react';
import './WeatherCard.css'


const WeatherCard = (props) => {

  // constImageUrl =
  return(
    <div className='weather-card'>
      <p>{props.weather.name}</p>
      <img src={require('../../resources/images/sun.png')}/>
      <p>{props.weather.currently.summary}</p>
    </div>
  );//closes return
}


export default WeatherCard;
