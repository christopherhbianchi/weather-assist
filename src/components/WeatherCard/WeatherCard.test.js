import React from 'react';
import { shallow } from 'enzyme';
import { WeatherCard } from './WeatherCard.js';

describe('<WeatherCard />', () => {

  it('should match snapshot', () => {
    const weather = { //weather object, currently object, summary is a propert of currently
      currently:{
        summary:'clear',
        temperature:70,
        windSpeed:10,
        precipProbability: 20
      }
    };//weather
    const renderedComponent = shallow(
      <WeatherCard
        weather={weather}
        removeCard={ () => {} } />
    );//shallow
    expect(renderedComponent).toMatchSnapshot();
  });//it


});//describe
