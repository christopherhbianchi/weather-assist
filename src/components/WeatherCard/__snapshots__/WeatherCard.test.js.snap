// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherCard /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeatherCard
    removeCard={[Function]}
    weather={
        Object {
            "currently": Object {
              "precipProbability": 20,
              "summary": "clear",
              "temperature": 70,
              "windSpeed": 10,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="remove-button"
          onClick={[Function]}
/>,
        <h4
          className="cityName"
/>,
        <img
          src="unknown.svg"
/>,
        <p>
          clear
</p>,
        <p>
          70
          ° F
</p>,
        <p>
          Wind Speed: 
          10
           mph
</p>,
        <p>
          Chance of Rain: 
          20
          %
</p>,
      ],
      "className": "weather-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "remove-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "cityName",
        },
        "ref": null,
        "rendered": null,
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "unknown.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "clear",
        },
        "ref": null,
        "rendered": "clear",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            70,
            "° F",
          ],
        },
        "ref": null,
        "rendered": Array [
          70,
          "° F",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Wind Speed: ",
            10,
            " mph",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Wind Speed: ",
          10,
          " mph",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Chance of Rain: ",
            20,
            "%",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Chance of Rain: ",
          20,
          "%",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="remove-button"
            onClick={[Function]}
/>,
          <h4
            className="cityName"
/>,
          <img
            src="unknown.svg"
/>,
          <p>
            clear
</p>,
          <p>
            70
            ° F
</p>,
          <p>
            Wind Speed: 
            10
             mph
</p>,
          <p>
            Chance of Rain: 
            20
            %
</p>,
        ],
        "className": "weather-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "remove-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "cityName",
          },
          "ref": null,
          "rendered": null,
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "unknown.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "clear",
          },
          "ref": null,
          "rendered": "clear",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              70,
              "° F",
            ],
          },
          "ref": null,
          "rendered": Array [
            70,
            "° F",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Wind Speed: ",
              10,
              " mph",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Wind Speed: ",
            10,
            " mph",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Chance of Rain: ",
              20,
              "%",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Chance of Rain: ",
            20,
            "%",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
